oss/oss.go

```diff
@@ -0,0 +1,779 @@
+	err := o.metaData(o.fs)
+	if err != nil {
+		fs.Logf(o, "Failed to read metadata: %v", err)
+		return time.Now()
+	}
+	// read mtime out of metadata if available
+	d, ok := o.meta[metaMtime]
+	if !ok || d == nil {
+		// fs.Debugf(o, "No metadata")
+		return o.lastModified
+	}
+	modTime, err := swift.FloatStringToTime(*d)
+	if err != nil {
+		fs.Logf(o, "Failed to read mtime from object: %v", err)
+		return o.lastModified
+	}
+	return modTime
+}
+
+// SetModTime sets the modification time of the local fs object
+// 设置修改时间
+func (o *Object) SetModTime(modTime time.Time) error {
+	err := o.metaData(o.fs)
+	if err != nil {
+		return err
+	}
+	o.meta = make(map[string]*string)
+	o.meta[metaMtime] = StringToPointer(swift.TimeToFloatString(modTime))
+	if o.size >= maxSizeForCopy {
+		fs.Debugf(o, "SetModTime is unsupported for objects bigger than %v bytes", fs.SizeSuffix(maxSizeForCopy))
+		return nil
+	}
+	// Guess the content type
+	mimeType := fs.MimeType(o)
+	// Copy the object to itself to update the metadata
+	key := o.fs.root + o.remote
+	sourceKey := o.fs.bucket + "/" + key
+	bucket, err := o.fs.c.Bucket(o.fs.bucket)
+	_, erro := bucket.CopyObject(key, key, oss.ContentType(mimeType),
+		oss.CopySource(o.fs.bucket, url.QueryEscape(sourceKey)),
+		oss.MetadataDirective(oss.MetaReplace))
+	return erro
+}
+
+// Storable raturns a boolean indicating if this object is storable
+// object是否可储存
+func (o *Object) Storable() bool {
+	return true
+}
+
+// Open an object for read
+// 获取一个对象实例
+func (o *Object) Open(options ...fs.OpenOption) (in io.ReadCloser, err error) {
+	key := o.fs.root + o.remote
+	bucket, _ := o.fs.c.Bucket(o.fs.bucket)
+	resp, err := bucket.GetObject(key)
+	if err != nil {
+		return nil, err
+	}
+	return resp, nil
+}
+
+// Update the Object from in with modTime and size
+// 使用metaData更新对象
+func (o *Object) Update(in io.Reader, src fs.ObjectInfo, options ...fs.OpenOption) error {
+	err := o.fs.Mkdir("")
+	if err != nil {
+		return err
+	}
+	mimeType := fs.MimeType(src)
+	bucket, err := o.fs.c.Bucket(o.fs.bucket)
+	key := o.fs.root + o.remote
+	erro := bucket.PutObject(key, in, oss.ContentType(mimeType))
+	if erro != nil {
+		return erro
+	}
+	// Read the metadata from the newly created object
+	o.meta = nil // wipe old metadata
+	erro = o.metaData(o.fs)
+	return erro
+}
+
+// Remove an object
+// 删除一个object
+func (o *Object) Remove() error {
+	bucket, _ := o.fs.c.Bucket(o.fs.bucket)
+	key := o.fs.root + o.remote
+	isObjectExist, _ := bucket.IsObjectExist(key)
+	if isObjectExist {
+		erro := bucket.DeleteObject(key)
+		return erro
+	} else {
+		return nil
+	}
+}
+
+// MimeType returns the content type of the Object if known, or "" if not
+// MimeType of an Object if known, "" otherwise
+// 返回object的类型
+func (o *Object) MimeType() string {
+	err := o.metaData(o.fs)
+	if err != nil {
+		fs.Logf(o, "Failed to read metadata: %v", err)
+		return ""
+	}
+	return o.mimeType
+}
+
+// String returns a pointer to the string value passed in.
```

Comment:
StringToPointer returns...
Why you define function for this operation. 
It`s standard go operation.